src/Apps/Order2/Routes/Details/Components/Order2DetailsFulfillmentInfo.tsx

```diff
@@ -16,10 +16,23 @@ export const Order2DetailsFulfillmentInfo: React.FC<
   const { fulfillmentDetails, selectedFulfillmentOption, shippingOrigin } =
     orderData
 
-  if (!selectedFulfillmentOption) return null
-
+  const isBlankFulfillmentDetails =
+    !fulfillmentDetails ||
+    Object.values(fulfillmentDetails).every(
+      part =>
+        !part ||
+        part === undefined ||
+        (typeof part === "string" && part.trim() === ""),
+    )
   const isPickup = selectedFulfillmentOption?.type === "PICKUP"
 
+  if (
+    (isPickup && !shippingOrigin) ||
+    (!isPickup && isBlankFulfillmentDetails)
+  ) {
+    return null
+  }
```

Comment:
Ah, maybe selectedFulfillmentOption doesn't show for all order fulfillment types yet (like arta)?

This seems right to me. I think you can do something like `if (part == null || // ...` instead of needing to check `!part || part === undefined` but that is really minor and not necessarily even better or more readable.